src/types/definitions.d.ts

```diff
@@ -8,3 +9,39 @@ export type EventConstructor<T extends DiscordEvent> = new (
 export type InteractionConstructor<T extends Interaction> = new (
```

Comment:
this looks like voodoo magic to me lol

Reply:
Ok yeah it's spooky voodoo shit.

I'm really bad at explaining this but I'll try. This took me a lot of effort to figure out

https://github.com/shinray/vythm/blob/482228caa91680b2ca498227f2aa8cb66cb1d85c/src/handlers/interactionHandler.ts#L29

See this, for example - we need a way to basically dynamically construct a class after reading files. 
everything in the interactions directory should be read into some type/extension of Interaction (base class)

https://github.com/shinray/vythm/blob/482228caa91680b2ca498227f2aa8cb66cb1d85c/src/handlers/eventHandler.ts#L24
same thing here. eventconstructor allows me to build generic Events based on whatever I read from the events directory

Reply:
TL;DR dynamically load and instantiate classes from files into code

Reply:
on client ready (and hopefully, on command later TODO) can dynamically load functionality into the bot